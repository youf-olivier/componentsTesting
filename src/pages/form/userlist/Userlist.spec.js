import React from 'react';
import { render, screen } from '@testing-library/react';

import UserList from './Userlist';

describe('Userlist tests suite', () => {
  const userList = [
    {
      id: 1,
      login: 'login1',
      avatar_url: 'http://avatar1.jpg',
      html_url: 'http://comptegithub.com/login1',
    },
    {
      id: 2,
      login: 'login2',
      avatar_url: 'http://avatar2.jpg',
      html_url: 'http://comptegithub.com/login2',
    },
    {
      id: 3,
      login: 'login3',
      avatar_url: 'http://avatar3.jpg',
      html_url: 'http://comptegithub.com/login3',
    },
  ];
  it('should render the list with user list full', () => {
    const { asFragment } = render(<UserList users={userList} />);
    expect(asFragment()).toMatchSnapshot(); // Pas forcément utile ici

    expect(screen.getByRole('list', { name: 'liste comptes' })).toBeInTheDocument();
    expect(screen.getAllByRole('listitem', { name: 'carte utilisateur' })).toHaveLength(3);
  });

  it('Should render message without user', () => {
    render(<UserList users={[]} />);
    expect(screen.getByText(/Aucun utilisateur trouvé/)).toBeInTheDocument();
    // ou plus simplement
    screen.getByText(/Aucun utilisateur trouvé/);
  });
});
