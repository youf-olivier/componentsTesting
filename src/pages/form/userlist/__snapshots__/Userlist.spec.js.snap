// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Userlist tests suite should render the list with user list full 1`] = `
<DocumentFragment>
  <div
    aria-label="liste comptes"
    class="user-list"
    role="list"
  >
    <div
      aria-label="carte utilisateur"
      class="user-card"
      role="listitem"
    >
      <a
        class="user-card__link"
        href="http://comptegithub.com/login1"
      >
        <img
          alt="login1"
          class="user-card__avatar"
          src="http://avatar1.jpg"
        />
        <span
          class="user-card__login"
        >
          login1
        </span>
      </a>
    </div>
    <div
      aria-label="carte utilisateur"
      class="user-card"
      role="listitem"
    >
      <a
        class="user-card__link"
        href="http://comptegithub.com/login2"
      >
        <img
          alt="login2"
          class="user-card__avatar"
          src="http://avatar2.jpg"
        />
        <span
          class="user-card__login"
        >
          login2
        </span>
      </a>
    </div>
    <div
      aria-label="carte utilisateur"
      class="user-card"
      role="listitem"
    >
      <a
        class="user-card__link"
        href="http://comptegithub.com/login3"
      >
        <img
          alt="login3"
          class="user-card__avatar"
          src="http://avatar3.jpg"
        />
        <span
          class="user-card__login"
        >
          login3
        </span>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
